mediapipe/tasks/ios/genai/inference/sources/LlmInference.swift

```diff
@@ -0,0 +1,143 @@
+      topk: options.topk,
+      temperature: options.temperature,
+      random_seed: options.randomSeed)
+    llmTaskRunner = LlmTaskRunner(sessionConfig: sessionConfig)
+
+    super.init()
+  }
+
+  /// A convenience initializer that creates a new instance of `LlmInference` from an absolute path
+  /// to a model asset bundle stored locally on the device and the default `LlmInference.Options`.
+  ///
+  /// - Parameters:
+  ///   - modelPath: The absolute path to a model asset bundle stored locally on the device.
+  @objc public convenience init(modelPath: String) {
+    let options = Options(modelPath: modelPath)
+    self.init(options: options)
+  }
+
+  /// Generates a response based on the input text.
+  ///
+  /// - Parameters:
+  ///   - inputText: A `String` that is used to query the LLM.
+  /// - Throws: An error if the LLM's response is invalid.
+  @objc public func generateResponse(inputText: String) throws -> String {
+    let tokens = try llmTaskRunner.predict(inputText: inputText)
+    guard let humanReadableLlmResponse = LlmInference.humanReadableString(llmResponses: tokens)
+    else {
+      throw LlmInferenceError.invalidResponseError
+    }
+
+    return humanReadableLlmResponse
+  }
+
+  private static func humanReadableString(
+    llmResponses: [String], stripLeadingWhitespaces: Bool = true
+  ) -> String? {
+    guard let llmResponse = llmResponses.first else {
+      return nil
+    }
+    return llmResponse.humanReadableString(stripLeadingWhitespaces: stripLeadingWhitespaces)
+  }
+
+}
+
+// Extension to `LlmInference` for defining `LlmInference.Options`
+extension LlmInference {
+  /// Options for setting up a `LlmInference`.
+  ///
+  /// Note: Inherits from `NSObject` for Objective C interoperability.
+  @objc(MPPLlmInferenceOptions) public final class Options: NSObject {
+    /// The absolute path to the model asset bundle stored locally on the device.
+    @objc public var modelPath: String
+
+    /// The total length of the kv-cache. In other words, this is the total number of input + output
+    /// tokens the model needs to handle.
+    @objc public var maxSequenceLength: Int = 512
+
+    /// The top K number of tokens to be sampled from for each decoding step. A value of 1 means
+    /// greedy decoding. Defaults to 40.
+    @objc public var topk: Int = 40
+
+    /// The randomness when decoding the next token. A value of 0.0f means greedy decoding. Defaults
+    /// to 0.8.
+    @objc public var temperature: Float = 0.8
+
+    /// The random seed for sampling tokens.
+    @objc public var randomSeed: Int = 0
+
+    /// Creates a new instance of `Options` with the modelPath and default values of
+    /// `maxSequenceLength`, `topK``, `temperature` and `randomSeed`.
+    /// This function is only intended to be used from Objective C.
+    ///
+    /// - Parameters:
+    ///   - modelPath: The absolute path to a model asset bundle stored locally on the device.
+    @objc public init(modelPath: String) {
```

Comment:
A more `Swift`y pattern is to provide an init with default params and declare the variables to be constants that can only be initialised once. But default params are not visible to Objective C. Hence adopting this pattern for a cleaner interface in both languages.